src/hooks/admin/useApiAccessTokensData.ts

```diff
@@ -0,0 +1,47 @@
+import { useEffect, useState, SetStateAction, Dispatch } from 'react';
+
+import { PermissionsWithAccessToken } from 'generated/sdk';
+import { useDataApi } from 'hooks/common/useDataApi';
+
+export function useApiAccessTokensData(): {
+  loadingApiAccessTokens: boolean;
+  apiAccessTokens: PermissionsWithAccessToken[];
+  setApiAccessTokensWithLoading: Dispatch<
+    SetStateAction<PermissionsWithAccessToken[]>
+  >;
+} {
+  const [apiAccessTokens, setApiAccessTokens] = useState<
+    PermissionsWithAccessToken[]
+  >([]);
+  const [loadingApiAccessTokens, setLoadingApiAccessTokens] = useState(true);
+
+  const api = useDataApi();
+
+  const setApiAccessTokensWithLoading = (
+    data: SetStateAction<PermissionsWithAccessToken[]>
+  ) => {
+    setLoadingApiAccessTokens(true);
+    setApiAccessTokens(data);
+    setLoadingApiAccessTokens(false);
+  };
+
+  useEffect(() => {
+    setLoadingApiAccessTokens(true);
+    api()
+      .getAllApiAccessTokensAndPermissions()
+      .then(data => {
+        if (data.allAccessTokensAndPermissions) {
+          setApiAccessTokens(
+            data.allAccessTokensAndPermissions as PermissionsWithAccessToken[]
+          );
+        }
+        setLoadingApiAccessTokens(false);
+      });
```

Comment:
Please here and in the other hook check if the component was unmounted before updating the state.